src/openms/source/DATASTRUCTURES/DataValue.cpp

```diff
@@ -198,7 +198,19 @@ namespace OpenMS
     }
   }
 
-  void DataValue::clear_()
+  DataValue::DataValue(DataValue&& rhs) noexcept :
+    value_type_(rhs.value_type_),
+    unit_type_(rhs.unit_type_),
+    unit_(rhs.unit_),
+    data_(rhs.data_)
+  {
+    // clean up rhs 
+    rhs.value_type_ = EMPTY_VALUE;
+    rhs.unit_type_ = OTHER;
+    rhs.unit_ = -1;
+  }
```

Comment:
this is a move constructor, the idea is that we simply "steal" the rhs stuff including all pointers and we store it in our object and then reset the rhs to an empty state. The object is now in a "a valid but unspecified state." 
https://stackoverflow.com/questions/7930105/does-moving-leave-the-object-in-a-usable-state
Note that similar to `clear_`, we still have the actual ptr / or data in `rhs.data_` and did not reset that

Reply:
I still consider myself a move newbie so I have a few questions. Is this move constructor needed? I would assume a synthesized one (that uses memmove internally) would even be faster? Cleaning up rhs seems to be additional overhead and from what I read not necessary.

Reply:
A simpler answer is here:  https://stackoverflow.com/questions/9965461/in-what-scenarios-should-i-expect-to-explicitly-need-to-implement-a-move-constru

Q: In what scenarios should I expect to explicitly need to implement a move constructor and move assignment 
A: If you find yourself implementing, any of:
-    destructor
-    copy constructor
-    copy assignment

Reply:
> Cleaning up rhs seems to be additional overhead and from what I read not necessary.

yes, I think we have a few alternatives. We could go for either

- `rhs.value_type_ = EMPTY_VALUE;`
- `rhs.data_ = nullptr;`

and we dont really need to reset `rhs.unit_type_` and `rhs.unit_` so we could remove that

Reply:
> Cleaning up rhs seems to be additional overhead and from what I read not necessary.

I dont think there is much overhead and it does put the object into a nice, well defined empty state so that it could even be used again. Also, one may even argue that all three values above are in one 64 bit block of space (8 bytes), on some architectures setting all three values may even be *faster* since you can set a single 64 value for the whole block -- if the compiler is smart enough :-) Not that you would ever notice the difference